rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -175,9 +185,15 @@ void SqliteStorage::fill_topics_and_types()
   }
 }
 
-std::string SqliteStorage::get_database_name(const std::string & uri)
+std::unique_ptr<rosbag2_storage::BagMetadata> SqliteStorage::load_metadata(const std::string & uri)
 {
-  return rosbag2_storage::FilesystemHelper::get_folder_name(uri) + ".db3";
+  try {
+    rosbag2_storage::MetadataIo metadata_io;
+    return std::make_unique<rosbag2_storage::BagMetadata>(metadata_io.read_metadata(uri));
+  } catch (std::exception & e) {
+    (void) e;
```

Comment:
Shouldn't there at least be some kind of logging?

Reply:
You're right, we'll do so.